app/models/subscription.rb

```diff
@@ -6,6 +6,7 @@ class Subscription < ApplicationRecord
 
   validates :starts_on, presence: true
   validate :starts_on_cannot_be_in_the_past
+  validate :unique_subscription
```

Comment:
Have you tried using the built-in uniqueness validation? See https://boringrails.com/tips/rails-unique-scope